src/main/scala/rocket/ALU.scala

```diff
@@ -45,51 +85,51 @@ object ALU
   def cmpEq(cmd: UInt) = !cmd(3)
 }
 
-import ALU._
-
-class ALU(implicit p: Parameters) extends CoreModule()(p) {
+abstract class AbstractALU(alufn: ALUFN)(implicit p: Parameters) extends CoreModule()(p) {
   val io = IO(new Bundle {
     val dw = Input(UInt(SZ_DW.W))
-    val fn = Input(UInt(SZ_ALU_FN.W))
+    val fn = Input(UInt(alufn.SZ_ALU_FN.W))
     val in2 = Input(UInt(xLen.W))
     val in1 = Input(UInt(xLen.W))
     val out = Output(UInt(xLen.W))
     val adder_out = Output(UInt(xLen.W))
     val cmp_out = Output(Bool())
   })
+}
 
+class ALU(fn: ALUFN)(implicit p: Parameters) extends AbstractALU(fn)(p) {
```

Comment:
I think this can just be `class ALU(implicit p: Parameters) extends AbstractALU(ALUFN())(p) {

Reply:
Done.